nntrainer/dataset/databuffer_factory.cpp

```diff
@@ -12,24 +12,30 @@
 
 #include <databuffer_factory.h>
 
-#include <databuffer_file.h>
-#include <databuffer_func.h>
+#include <data_producers.h>
+#include <func_data_producer.h>
 #include <nntrainer_error.h>
+#include <raw_file_data_producer.h>
 
 namespace nntrainer {
 
 /**
  * @brief Factory creator with constructor
  */
 std::unique_ptr<DataBuffer> createDataBuffer(DatasetType type) {
+
+  std::unique_ptr<DataProducer> dp;
   switch (type) {
   case DatasetType::FILE:
-    return std::make_unique<DataBufferFromDataFile>();
+    dp = std::make_unique<RawFileDataProducer>();
+    break;
```

Comment:
we just create DatasetType::FILE without file path?

Reply:
Yes, for the file, there are two way of creating it.

1. without file path, and later set as a property
2. with file path, (can be overriden with a property)